boards: mps3_an547: Enable QEMU support

galak opened the pull request:
Now that SDK 0.13.0 is out we can enable QEMU support on the
MPS3-AN547 to get coverage on Cortex-M55.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
boards: mps3_an547: Enable QEMU support

Now that SDK 0.13.0 is out we can enable QEMU support on the
MPS3-AN547 to get coverage on Cortex-M55.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

microbuilder reviewed (approved):
Tested with:

```bash
$ west build -p auto -b mps3_an547 zephyr/samples/hello_world -t run
*** Booting Zephyr OS build zephyr-v2.6.0-1922-g627cc921e422  ***
Hello World! mps3_an547
```

stephanosio committed:
boards: arm: mps3_an547: Disable null pointer detection for QEMU

This commit disables the Cortex-M null pointer detection feature for
the QEMU mps3_an547 targets because QEMU permits bus access to the
unmapped 0x0-0x400 region used by the MPU-based null pointer detection
feature.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>

galak commented on boards/arm/mps3_an547/Kconfig.defconfig:
```diff
@@ -6,6 +6,14 @@ if BOARD_MPS3_AN547
+# MPU-based null-pointer dereferencing detection cannot
+# be applied as the (0x0 - 0x400) is unmapped but QEMU
+# will still permit bus access.
+choice NULL_POINTER_EXCEPTION_DETECTION
```
Is this a QEMU bug?

stephanosio replied:
This was actually copied off the `mps2_an521` board:

https://github.com/zephyrproject-rtos/zephyr/blob/f3435b75efbfd7614ff873f8bfa90a031d6d27e8/boards/arm/mps2_an521/Kconfig.defconfig#L6-L12

The Cortex-M null pointer detection (MPU mode) relies on the MPU to generate faults when the 0x0-0x400 "page" is accessed, and it does not do that in case of QEMU according to the comment (I haven't tested it on FVP).

https://github.com/zephyrproject-rtos/zephyr/blob/b3b36f69a6444baac1cad1a67590e5fd7af70cf4/arch/arm/core/aarch32/mpu/arm_mpu.c#L356-L373

stephanosio committed:
tests: ztest: error_hook: Skip divide-by-zero fault test for QEMU MPS3

QEMU does not generate divide-by-zero exception on the MPS3 emulation,
so do not test it.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>
